src/stores/my-account.ts

```diff
@@ -5,20 +5,42 @@ import {
   loginWithSecretKey,
   Signer,
 } from '@/utils/account'
+import { getWeb3AuthClientId } from '@/utils/env/client'
+import { generateRandomName } from '@/utils/random-name'
 import { LocalStorage } from '@/utils/storage'
+import { SubstrateRPC } from '@/utils/substrate-rpc'
+import * as bottts from '@dicebear/bottts'
+import { createAvatar } from '@dicebear/core'
+import { CHAIN_NAMESPACES, SafeEventEmitterProvider } from '@web3auth/base'
+import { Web3Auth } from '@web3auth/modal'
 import { useAnalytics } from './analytics'
 import { create } from './utils'
 
+const clientId = getWeb3AuthClientId()
+
+type AuthenticatedUser = {
+  name: string
+  encodedSecretKey: string
+  address: string
+  profilePic: string
+  email?: string
+  username?: string
+}
+
 type State = {
-  isInitialized?: boolean
-  isInitializedAddress?: boolean
+  web3Auth: Web3Auth | null
+  authenticatedUser: AuthenticatedUser | null
+  provider: SafeEventEmitterProvider | null
```

Comment:
this web3auth and provider, you only need this to be run once right? if so, then it doesn't need to be inside the state, just put it as global variable where the intiial is null,

and when you login, use singleton pattern to check if its already init (like you have done here), and just assign the initialized data to the global variable.

Reply:
I think we need to have the web3Auth in store. It is currently used in different actions like login, logout and also getting the private key

Since we'll be using web3auth, we need the same initialized instance of the web3auth which can only be from the store. If I make it a global variable, then the value wouldn't be preserved would it?

I will remove the provider from store though, it can always be gotten by doing `web3auth.provider`